### ISSUE TITLE
Ability to upload app bundles to App Center

### ISSUE THREAD

**Issue by rs146** [microsoft/appcenter#1288]
We having been trying to manually upload an aab (Android app Bundle) to the Releases section of our app however it seems that App Center only allows APK uploads. Our question is this: Is there a means to upload an Android app Bundle to App Center?

![Screenshot 2019-10-31 at 10 02 43](https://user-images.githubusercontent.com/2216358/67937532-a95bb700-fbc5-11e9-96df-ee53d383d3fb.png)

**Comment by Oddj0b** [comment]
Hi @rs146, sorry to say it's not possible to upload .aab files without releasing it to a Google Store. You can't upload directly to distribution groups hence issue #955. I suggest you give the mentioned issue a 👍 to make it more likely to be get picked-up.

**Comment by Simulacrux** [comment]
So Google also send out this message in the PlayStore:

> Starting August 2021, new apps will be required to target API level 30 (Android 11) and use the Android App Bundle publishing format. Starting November 2021, all app updates will be required to target API level 30 (Android 11).

As it a bit unclear if the .abb files also start in November for all apps, but time will run out for this issue.

**Comment by duenes25** [comment]
Google Play store will not be taking .apk format for new apps.  Having the ability to upload .aab files into AppCenter is no longer a luxury.  Also, google has announced that Firebase App Distribution is supporting .aab format.  Can this issue get higher priority?

**Comment by mcskik** [comment]
Uploading *.San Android app bundle files to MS App Center for distribution to QA testers outside of the Google play store is an essential feature otherwise we will be forced to ditch our use of MS App Center and find a better tool for our company.

**Comment by Brantone** [comment]
With the previous upload API we were able to upload AAB files because the app OS was set to "custom" and it allowed it.
Now with the "encouragement" to use appcenter CLI, it throws an error that AAB is not supported, despite it being set to custom.
So not only is AAB _still_ not officially supported, but the one option to compromise has now been removed.

**Comment by nirmal-mobiquity** [comment]
deadline is very near for playstore to support aab and we  don't want to have two different bundle type, our code is generating both bundle type but not having support to upload aab till now is demotivating.

**Comment by mohamed95ivy** [comment]
@botatoes, @dimazaid Any updates on this issue?

**Comment by Brantone** [comment]
AAB isn't going away any time soon.

**Comment by quentinR0bert** [comment]
Any news about this ? Because we have a very strange repro from AppCenter.

For some days we have see that AppCenter allow the upload of AppBundle for the Distribute part, so we were like "yes ! let's go ! Let's try and remove BundleTool from our CI if it's okay".

When we upload manually a AppBundle, it's okay and when we go to the url with a phone when we download we get an apk (an universal we suppose).

And when we upload from appcenter-cli it's okay, but when we go to the url with a phone when we download we only get the AppBundle file ! Why ?

**Comment by Talgat777** [comment]
Ping! So that this issue is not closed automatically!

**Comment by thomaskaelin** [comment]
Arrivedin 2022, still no official comment on this? Seriously guys, not having this feature by now is just ridiculous. Can't be that hard?

**Comment by boliveira** [comment]
I have just tested this and AppCenter does support uploading aabs and downloading both the aab and apk. Am I missing something?

**Comment by Brantone** [comment]
It looks to work via UI; however, using the CLI, returns:
```
{"succeeded":false,"errorCode":4,"errorMessage":"Files of type '.aab' can not be distributed to groups"}
```

**Comment by icatstudios** [comment]
One more ping from my side!
This must be the priority. Even .apks should be deprecated and removed as Google left it alone already.

**Comment by norman-kapschefsky** [comment]
Via Web-UI I was able to upload an aab without any problems. Still try to find a proper android gradle plugin that supports that kind of aab upload ...

![image](https://user-images.githubusercontent.com/813760/153159493-694f2a04-b6fd-431d-833f-d443d19062a0.png)
![image](https://user-images.githubusercontent.com/813760/153159552-5e0095fb-ba7e-455c-9aa5-e1af24eda456.png)

**Comment by boliveira** [comment]
I use fastlane with the appcenter plugin and it works perfectly for both aabs and apks. Just make sure that when uploading an aab, you upload it to the Play Store as they cannot be distributed to collaborators (Although you can download an apk from the aab in AppCenter).

**Comment by stantronic** [comment]
Any news on this? Trying to write a script to upload an .aab using the API docs - just get 400 error

**Comment by AndreasHennig** [comment]
As stated by @m-benamar, one workaround to the problem described here is to generate a universal APK from the AAB using a tool called `bundletool`. Since this is actually the same tool Google uses to generate the device-optimized APKs after you upload the AAB to Google Play, for me this doesn't feel that "wrong".

After looking into it I discovered that the process actually isn't that complicated: 
1. You build your AAB
2. generate what is called an APK set using the `bundletool build-apks` command
3. and extract the universal APK from the APK set

I wrote a blog post where I describe the process in greater detail: https://www.andreashennig.com/blog/generating-universal-apk-from-aab/

Maybe this helps some of you overcoming this current limitation.
